golden_hammer/module.go

```diff
@@ -0,0 +1,275 @@
+			}
+			continue
+		}
+		if containsService(*document, types.BaseURLServiceType) {
+			h.fixedDocumentDIDs[id.String()] = true
+			continue
+		}
+		// This document needs fixing
+		documents = append(documents, *document)
+	}
+	// Sort: since care organization DIDs refer to vendor DIDs through NutsComm service,
+	// vendor DIDs should be fixed first. Meaning DIDs without NutsComm service should go first.
+	sort.Slice(documents, func(i, j int) bool {
+		return containsService(documents[i], transport.NutsCommServiceType)
+	})
+	return documents, nil
+}
+
+func (h *GoldenHammer) tryResolveURL(id did.DID) (*url.URL, error) {
+	// DIDIdentifierResolver only looks at DID document to resolve OpenID4VCI Identifiers.
+	didIDResolver := oidc4vci.DIDIdentifierResolver{ServiceResolver: didservice.NewServiceResolver(h.docResolver)}
+	// TLSIdentifierResolver looks at TLS certificate to resolve OpenID4VCI Identifiers.
+	tlsIDResolver := oidc4vci.NewTLSIdentifierResolver(didIDResolver, h.tlsConfig)
+
+	// Check if the services base URL is present
+	identifier, err := didIDResolver.Resolve(id)
+	if err != nil {
+		return nil, err
+	}
+	if identifier == "" {
+		// Not registered, try to resolve it using the TLS certificate
+		identifier, err = tlsIDResolver.Resolve(id)
+		if err != nil {
+			return nil, err
+		}
+		if identifier == "" {
+			return nil, nil
+		}
+	}
+	// Identifier is: baseURL + /n2n/identity/<did>
+	return url.Parse(identifier[:strings.Index(identifier, "/n2n/identity/")])
+}
+
+// resolveContainsService returns whether 1. given DID document can be resolved, and 2. it contains the specified service.
+func (h *GoldenHammer) resolveContainsService(id did.DID, serviceType string) bool {
+	document, _, err := h.docResolver.Resolve(id, nil)
+	if didservice.IsFunctionalResolveError(err) {
+		// Unresolvable DID document, nothing to do
+		return false
+	}
+	if err != nil {
+		// Other error occurred
+		log.Logger().WithError(err).Infof("Can't resolve DID document, skipping fix (did=%s)", id)
+		return false
+	}
+	return containsService(*document, serviceType)
+}
+
+func getService(document did.Document, serviceType string) *did.Service {
+	for _, service := range document.Service {
+		if service.Type == serviceType {
+			return &service
+		}
+	}
+	return nil
+}
+
+// getServiceEndpoint returns the endpoint of the given service as string, if present.
+// If the document does not contain a service with the given type or it isn't a string, an empty string is returned.
+func getServiceEndpoint(document did.Document, serviceType string) string {
+	service := getService(document, serviceType)
+	if service == nil {
+		return ""
+	}
+	var endpoint string
+	_ = service.UnmarshalServiceEndpoint(&endpoint)
+	return endpoint
+}
+
+func containsService(document did.Document, serviceType string) bool {
+	service := getService(document, serviceType)
+	return service != nil
+}
```

Comment:
There's a similar function `ByServiceType` (matcher) which has same behavior, but using a matcher-type function is kinda weird here (although it works) code-wise, and `ByServiceType` is not explicit about matching references or not (it's just a comment), so prone to bugs in future, when it changes.

Moving it there doesn't make much sense either, since we plan to remove this module as soon as it isn't required any more, making the util funcs moved to `didservice` dead code.